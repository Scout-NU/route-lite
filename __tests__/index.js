import React from 'react';
import renderer from 'react-test-renderer';
import Router, { goTo, goBack } from '../index';

const A = () => <div>Component A</div>;

test('Router renders default component only when nothing is on the stack', () => {
  const component = renderer.create(
    <Router>
      <div>Hello, world!</div>
    </Router>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  goTo(A);
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  goBack();
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
