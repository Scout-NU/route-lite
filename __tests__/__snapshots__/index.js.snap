// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Links handle clicks properly 1`] = `
<a
  className=""
  href="http://google.com"
  id=""
  onClick={[Function]}
>
  Click here!
</a>
`;

exports[`Links handle clicks properly 2`] = `
<div>
  Next Component
</div>
`;

exports[`Renders plain links properly 1`] = `
<a
  className="my-class"
  data-prop="42"
  href="http://google.com"
  id="the-link"
  onClick={[Function]}
>
  Click here!
</a>
`;

exports[`Router renders default component only when nothing is on the stack 1`] = `
<div>
  Hello, world!
</div>
`;

exports[`Router renders default component only when nothing is on the stack 2`] = `
<div>
  Component A
</div>
`;

exports[`Router renders default component only when nothing is on the stack 3`] = `
<div>
  Hello, world!
</div>
`;
